app/src/test/java/com/x8bit/bitwarden/data/platform/repository/ServerConfigRepositoryTest.kt

```diff
@@ -0,0 +1,199 @@
+import com.x8bit.bitwarden.data.platform.datasource.network.service.ConfigService
+import com.x8bit.bitwarden.data.platform.manager.dispatcher.DispatcherManager
+import com.x8bit.bitwarden.data.platform.repository.model.Environment
+import com.x8bit.bitwarden.data.platform.repository.util.FakeEnvironmentRepository
+import com.x8bit.bitwarden.data.platform.util.asSuccess
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneOffset
+
+class ServerConfigRepositoryTest {
+    private val fakeDispatcherManager: DispatcherManager = FakeDispatcherManager()
+    private val fakeConfigDiskSource = FakeConfigDiskSource()
+    private val configService: ConfigService = mockk {
+        coEvery {
+            getConfig()
+        } returns CONFIG_RESPONSE_JSON.asSuccess()
+    }
+
+    private val environmentRepository = FakeEnvironmentRepository().apply {
+        environment = Environment.Us
+    }
+
+    private val fixedClock: Clock = Clock.fixed(
+        Instant.parse("2023-10-27T12:00:00Z"),
+        ZoneOffset.UTC,
+    )
+
+    private val repository = ServerConfigRepositoryImpl(
+        configDiskSource = fakeConfigDiskSource,
+        configService = configService,
+        clock = fixedClock,
+        environmentRepository = environmentRepository,
+        dispatcherManager = fakeDispatcherManager,
+    )
+
+    @BeforeEach
+    fun setUp() {
+        fakeConfigDiskSource.serverConfig = null
+    }
+
+    @Test
+    fun `environmentRepository stateflow should trigger new server configuration`() = runTest {
+        assertNull(
+            fakeConfigDiskSource.serverConfig,
+        )
+
+        // This should trigger a new server config to be fetched
+        environmentRepository.environment = Environment.Eu
+
+        repository.serverConfigStateFlow.test {
+            val newConfig = awaitItem()
+            assertNotNull(
+                newConfig,
+            )
+
+            assertEquals(
+                SERVER_CONFIG,
+                newConfig,
+            )
+        }
+    }
+
+    @Test
+    fun `getServerConfig should fetch a new server configuration with force refresh as true`() =
+        runTest {
+            coEvery {
+                configService.getConfig()
+            } returns CONFIG_RESPONSE_JSON.copy(version = "NEW VERSION").asSuccess()
+
+            fakeConfigDiskSource.serverConfig = SERVER_CONFIG.copy(
+                lastSync = fixedClock.instant().toEpochMilli(),
+            )
+
+            assertNotNull(
+                fakeConfigDiskSource.serverConfig,
```

Comment:
We shouldn't need all these `assertNotNull`s since the `assertEquals` will catch all that